packages/template-retail-react-app/app/hooks/use-pickup-shipment.js

```diff
@@ -0,0 +1,184 @@
+     * @param {string} basketId - The basket ID
+     * @param {string} shippingMethodId - The shipping method ID to set
+     * @param {boolean} throwOnError - Whether to throw on error (default: false)
+     */
+    const configureRegularShippingMethod = async (
+        basketId,
+        shippingMethodId,
+        throwOnError = false
+    ) => {
+        try {
+            await updateShipmentForBasketMutation.mutateAsync({
+                parameters: {
+                    basketId,
+                    shipmentId: 'me'
+                },
+                body: {
+                    shippingMethod: {
+                        id: shippingMethodId
+                    }
+                }
+            })
+        } catch (error) {
+            if (throwOnError) {
+                throw error
+            } else {
+                // Log error but don't block the add to cart flow
+                console.warn('Failed to configure regular shipping method:', error)
+            }
+        }
+    }
+
+    /**
+     * Checks if any items in the selection require pickup configuration
+     * @param {Array} productSelectionValues - Array of product selection values
+     * @param {Object} pickupInStoreMap - Map of product IDs to pickup flags
+     * @param {Object} mainProduct - Main product object (for fallback)
+     * @returns {boolean} True if any items are pickup items
+     */
+    const hasPickupItems = (productSelectionValues, pickupInStoreMap, mainProduct) => {
+        return productSelectionValues.some((item) => {
+            const prodKey =
+                (item.variant || item.product || mainProduct).productId ||
+                (item.variant || item.product || mainProduct).id
+            return pickupInStoreMap[prodKey]
+        })
+    }
+
+    /**
+     * Adds inventory ID to product items that have pickup selected
+     * @param {Array} productItems - Array of product items
+     * @param {Object} pickupInStoreMap - Map of product IDs to pickup flags
+     * @param {Object} storeInfo - Store information object containing inventoryId
+     * @returns {Array} Updated product items with inventory IDs
+     */
+    const addInventoryIdsToPickupItems = (productItems, pickupInStoreMap, storeInfo) => {
+        if (!storeInfo?.inventoryId) return productItems
+
+        return productItems.map((item) => {
+            const prodKey = item.productId || item.id
+            if (pickupInStoreMap[prodKey]) {
+                return {
+                    ...item,
+                    inventoryId: storeInfo.inventoryId
+                }
+            }
+            return item
+        })
+    }
+
+    return {
+        configurePickupShipment,
+        configureRegularShippingMethod,
+        hasPickupItems,
+        addInventoryIdsToPickupItems,
+        getPickupShippingMethodId,
+        getDefaultShippingMethodId,
+        isCurrentShippingMethodPickup,
+        isLoading: updateShipmentForBasketMutation.isLoading
```

Comment:
How about we return all of the `updateShipmentForBasketMutation` instead? There are a lot of methods return from `updateShipmentForBasketMutation` which is the mutation result of react query. only return isLoading here my result in missing other return when we need to use any of them.